src/top-navigation/styles.scss

```diff
@@ -60,9 +60,11 @@
   min-width: 0;
 
   > .identity-link {
-    display: flex;
+    display: grid;
+    grid-template-columns: max-content auto;
     align-items: center;
     text-decoration: none;
+    gap: awsui.$space-s;
```

Comment:
Is there a way to do the same without changing `display`?

Reply:
I tried fixing this with the current `display: flex` but there is too many magic to maintain like for example the `min-width: 1px` we currently have and the fact that if a customer provided an `svg` with 100% height and width the flex container will not know how much width/height should it give to the svg and the text always overflows.

Reply:
I got it working by applying `flex-grow: 1` to the `.identity` selector.

I think your solution with `display: grid` does something similar, but in more magical way, because this is how nested `display: grid` and `display: flex` interact.

IMO, having two blocks (logo and the text) in this placement is not a valid use-case for grid display

Reply:
that does not solve it tho, if I applied the same logic with an svg image of 100% height and width it'll still show the same issue.